vql/linux/audit.go

```diff
@@ -37,15 +43,63 @@ func (self *streamHandler) outputMultipleMessages(msgs []*auparse.AuditMessage)
 	self.output_chan <- event
 }
 
+type _AuditPluginArgs struct {
+        Rules      []string            `vfilter:"optional,field=rules,doc=List of rules"`
+}
+
 type AuditPlugin struct{}
 
+
 func (self AuditPlugin) Info(scope vfilter.Scope, type_map *vfilter.TypeMap) *vfilter.PluginInfo {
 	return &vfilter.PluginInfo{
 		Name: "audit",
 		Doc:  "Register as an audit daemon in the kernel.",
 	}
 }
 
+func contains(r [][]byte, r1 []byte) bool {
+	for _, v := range r {
+		if bytes.Equal(v, r1) {
+			return true
+		}
+	}
+	return false
+}
+
+func addRules(c *libaudit.AuditClient, rulelst []string, chkrules [][]byte, s vfilter.Scope) {
+       for _, line := range rulelst {
+              r, err := flags.Parse(line)
+              if err != nil {
+                     s.Log("Error parsing Rule %v: %s", err)
+                     return
+               }
+               data, err := rule.Build(r)
+               if err != nil {
+                      s.Log("Error Building Rule: %s", err)
+                      return
+               }
+               if len(chkrules) > 0 {
+                       if contains(chkrules, []byte(data)) == true{
+                             s.Log("Rule %s already exists", line)
+                             continue
+                       }
+               }
+               defer c.GetRules()
+               if err := c.AddRule([]byte(data)); err != nil {
+                       s.Log("Error while adding Rule %s: %s", line,err)
+                       return
+               }
+               auditrules = append(auditrules, []byte(data))
+               s.Log("Added Rule %s", line)
+       }
+
+}
+
+func deleteRules(client *libaudit.AuditClient) {
+	if _, err := client.DeleteRules(); err != nil {
```

Comment:
What happens if we have multiple queries that use the same rule? Are these considered separate because of the tagging in the query? Do separate tagged rules work properly?

Reply:
2 artifacts for User Account creation and Group creation has same rule with vrr_etc_nsswitch_conf and if the rule is added with the artifact which is executed first, it will not be loaded by other artifact as the Add function doesn't let u add the existing rule